### PR TITLE
Core/Quest: Threat From Above

### LINKED ISSUE #26355: [Quest] Threat From Above
**How it now**
When you start hitting the dragon, you leave him about 30-40% health, the NPCs (Cultist Bombardier 3x)  that sit on the dragon get off it and disappear 
sometimes they stay to fight
sometimes they run away and disappear 

**How it should be**
Cultist Bombardier shouldn't run away or disappear 

I think it has something to do with the starting point of the dragon's flight. The farther the dragon flies away from the starting point, the greater the chance to reproduce the bug 

[https://www.wowhead.com/quest=13809/threat-from-above](https://www.wowhead.com/quest=13809/threat-from-above)

3.3.5
rev. 80f30d5

### PR DESCRIPTION
**Pull request by Edder**
**Changes proposed:**

**Fix for Quest (13682) Threat From Above**

**First issue:**
- Two spells (60603 and 52205) Chillmaw is using to eject passengers from itself (a vehicle) missing EffectValues, EffectValues are used to determine the seatId to eject from
- Therefore the ejection wasn't working and two of the three npcs were ejected after Chillmaws death (wrong)

**Fix first issue:**
- Those two spells are now generic scripted in a unique way to set the seatId correctly and eject the passenger
- spell script "spell_eject_passenger_wild_wyrm" removed because the functionality is now implemented through generic script "spell_gen_eject_passenger_1"

**Second issue:**
- Chillmaw passengers share the same home position as Chillmaw
- Chillmaw moves on waypoints and if the passengers are ejected while Chillmaw is too far away from home, the npcs are too
- The npc won't attack the player and eventually just run away and evade

**Fix second issue:**
- Setting the home position of an ejected passenger (if creature) to its current positon

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:**

Closes #26355

**Tests performed:**

Build tested, in-game test

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

File: src/server/game/Entities/Creature/Creature.h [hunk 1]
```diff
@@ -372,6 +372,8 @@ class TC_GAME_API Creature : public Unit, public GridObject<Creature>, public Ma
 
         std::string GetDebugInfo() const override;
 
+        void ExitVehicle(Position const* exitPosition = nullptr);
```
Comment (illfated) [inline src/server/game/Entities/Creature/Creature.h:375]:
```cpp
/src/server/game/Entities/Creature/Creature.h:375:14:
 fatal error: 'ExitVehicle' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
        void ExitVehicle(Position const* exitPosition = nullptr);
             ^
```

File: sql/updates/world/3.3.5/2021_04_24_00_world.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+-- delete non generic spell script for wild wyrm
+DELETE FROM `spell_script_names` WHERE `ScriptName`= 'spell_eject_passenger_wild_wyrm';
```
Comment (illfated) [inline sql/updates/world/3.3.5/2021_04_24_00_world.sql:2]: Looks harmless enough to begin with, but this spell is actually used in Storm Peaks scripts:

https://github.com/TrinityCore/TrinityCore/blob/d5f1fefe7f6772b11e9fbb052bfd36be9bda98e2/src/server/scripts/Northrend/zone_storm_peaks.cpp#L1070-L1100

https://github.com/TrinityCore/TrinityCore/blob/d5f1fefe7f6772b11e9fbb052bfd36be9bda98e2/sql/old/3.3.5a/world/62_2017_04_18/2017_02_25_00_world.sql#L20-L33

If you remove that spell ScriptName without restoring a new spell ScriptName, you will find that [Quest 13003, **Thrusting Hodir's Spear**](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?quest=13003/thrusting-hodirs-spear), becomes a problem.
- https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?quest=13003/thrusting-hodirs-spear

Comment (offl) [reply sql/updates/world/3.3.5/2021_04_24_00_world.sql:2]: Check changes

**Comment by offl** [comment]
You can also describe changes you made in order to fix the quest(so PR will be reviewed faster)

**Comment by illfated** [comment]
OK, so I forgot to read the entire PR content. Sorry about that. Would it be OK to ask for a NewLine at the end of the SQL file, just to be on the safe side?

File: src/server/game/Entities/Creature/Creature.h (hunk 1 above)
Comment (Edder) [reply src/server/game/Entities/Creature/Creature.h]: Ups, looks like Visual Studio isn't capable of that.
https://developercommunity.visualstudio.com/t/add-a-warning-if-override-keyword-is-not-used-in-d/396484
I fixed it.

**Comment by Edder** [comment]
> You can also describe changes you made in order to fix the quest(so PR will be reviewed faster)
Edited the PR comment

**Comment by Edder** [comment]
> OK, so I forgot to read the entire PR content. Sorry about that. Would it be OK to ask for a NewLine at the end of the SQL file, just to be on the safe side?
> 
> edit: SQL codestyle/consistency: Would be nice to have the comment lines before the queries, the same way as on line 1, just to keep it consistent.
>
Done

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**